api/bmc/v1beta1/virtualmachinebmc_types.go

```diff
@@ -30,16 +30,34 @@ const (
 
 // VirtualMachineBMCSpec defines the desired state of VirtualMachineBMC.
 type VirtualMachineBMCSpec struct {
-	// Reference to the VM to manage.
-	VirtualMachineRef *corev1.LocalObjectReference `json:"virtualMachineRef,omitempty"`
+	// BMC Service configuration
+	// +optional
+	Service *BMCServiceSpec `json:"service,omitempty"`
 
-	// Reference to the Secret containing IPMI/Redfish credentials.
-	AuthSecretRef *corev1.LocalObjectReference `json:"authSecretRef,omitempty"`
+	// Reference to the Secret containing IPMI/Redfish credentials
+	// +Required
+	AuthSecretRef *corev1.LocalObjectReference `json:"authSecretRef"`
+
+	// Reference to the VM to manage
+	// +Required
+	VirtualMachineRef *corev1.LocalObjectReference `json:"virtualMachineRef"`
+}
+
+// Service configuration for the BMC service.
+type BMCServiceSpec struct {
+	// Type of service (LoadBalancer, NodePort, ClusterIP, etc)
+	Type corev1.ServiceType `json:"type,omitempty"`
```

Comment:
We can leverage the [validation markers](https://book.kubebuilder.io/reference/markers/crd-validation) to protect the field, such as giving it a default value (`ClusterIP`), and restrict the available values.